src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,518 @@
+                    final GeoJsonReader reader = new GeoJsonReader(new StringResource(string));
+                    while (reader.hasNext())
+                    {
+                        final PropertiesLocated propertiesLocated = reader.next();
+                        if (propertiesLocated.getItem() instanceof Polygon)
+                        {
+                            polygons.add((Polygon) propertiesLocated.getItem());
+                        }
+                        else
+                        {
+                            logger.warn("Polygon Filter does not support item {}",
+                                    propertiesLocated.toString());
+                        }
+                    }
+                    return Optional.of(polygons).filter(list -> !list.isEmpty());
+                };
+            case WKT:
+                return string -> Optional.of(Collections
+                        .singletonList(new WktPolygonConverter().backwardConvert(string)));
+            case WKB:
+                return string -> Optional.of(Collections.singletonList(
+                        new WkbPolygonConverter().backwardConvert(WKBReader.hexToBytes(string))));
+            case UNSUPPORTED:
+            default:
+                logger.warn("No converter set up for {} format. Supported formats are {}", format,
+                        Arrays.copyOf(PolygonStringFormat.values(),
+                                PolygonStringFormat.values().length - 1));
+                return string -> Optional.empty();
+        }
+    }
+
+    private static List<Polygon> getPolygonsFromFormatMap(
+            final Map<String, List<String>> polygonLists)
+    {
+        if (polygonLists.isEmpty())
+        {
+            return Collections.emptyList();
+        }
+        return polygonLists.entrySet().stream()
+                .flatMap(formatAndPolygonStrings -> formatAndPolygonStrings.getValue().stream().map(
+                        getPolygonConverterForFormat(formatAndPolygonStrings.getKey())::convert)
+                        .filter(Optional::isPresent).map(Optional::get).flatMap(List::stream))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns predicate that filters out MultiPolygons that overlap with any Polygons or outer
+     * MultiPolygons in the provided QuadTree. This also adds any non-intersecting MultiPolygon to
+     * the quad tree.
+     *
+     * @param vettedPolygons
+     *            non-overlapping polygons and multipolygons
+     * @param polygonType
+     *            whether this is an include or exclude Polygon
+     * @return filter for multipolygons
+     */
+    private static Predicate<MultiPolygon> notOverlappingMultipolygon(
+            final QuadTree<Located> vettedPolygons, final Type polygonType)
+    {
+        return multiPolygon ->
+        {
+            if (vettedPolygons.get(multiPolygon.bounds()).stream()
```

Comment:
Here we are iterating over the stream twice. Not sure how much of an impact on runtime this would make. Do you think these can be combined into one iteration?

Maybe have a predicate which checks for instance type and ```multipolygon overlaps```?

Reply:
It probably has minimal impact on runtime performance because this happens once during initialization and the quad tree is optimized for these type of spacial queries, and because only not overlapping polygons are stored in the tree, not all polygons will be iterated over.

That being said, it is much better design and practice to make this into a unified ```Predicate<Located>```, however I will need to make two for both Polygons and MultiPolygons which will increase the code duplication that I know is bothering @AlexHsieh22

Reply:
haha I'm working on getting over it :) also thinking of ways to simplify this more! Overall, very grateful for the work being done here!

Reply:
I was wrong. I was able to avoid too much duplicated code by nulling inputs in a more general function. The duplicated code I was less able to avoid is in PolygonStringFormat enum now.